packages/contracts/deploy/100_install_on_management_dao/100_install.ts

```diff
@@ -0,0 +1,315 @@
+    [ethers.constants.AddressZero, 0], // [target, operation]
+    '0x', // metadata
+  ]);
+
+  const prepareTx = await pspContract.prepareInstallation(
+    managementDAO.address,
+    {
+      pluginSetupRef,
+      data,
+    },
+    {
+      gasLimit: 30000000,
+    }
+  );
+
+  // extract info from prepare event
+  const event =
+    await findEventTopicLog<PluginSetupProcessorEvents.InstallationPreparedEvent>(
+      await prepareTx.wait(),
+      PluginSetupProcessor__factory.createInterface(),
+      'InstallationPrepared'
+    );
+
+  const installationPreparedEvent = event.args;
+
+  console.log(
+    `Prepared (Multisig: ${installationPreparedEvent.plugin} version (release: ${VERSION.release} / build: ${VERSION.build}) to be applied on (ManagementDAO: ${managementDAO.address}), see (tx: ${prepareTx.hash})`
+  );
+
+  // grant
+  // ROOT_PERMISSION on the management dao to the PSP
+  // APPLY_INSTALLATION_PERMISSION on the PSP to the deployer
+  const permissionsToGrant: DAOStructs.MultiTargetPermissionStruct[] = [
+    {
+      operation: Operation.Grant,
+      where: managementDAO.address,
+      who: pspAddress,
+      condition: ethers.constants.AddressZero,
+      permissionId: DAO_PERMISSIONS.ROOT_PERMISSION_ID,
+    },
+    {
+      operation: Operation.Grant,
+      where: pspAddress,
+      who: deployer.address,
+      condition: ethers.constants.AddressZero,
+      permissionId:
+        PLUGIN_SETUP_PROCESSOR_PERMISSIONS.APPLY_INSTALLATION_PERMISSION_ID,
+    },
+  ];
+
+  await managementDAO.applyMultiTargetPermissions(permissionsToGrant);
+
+  // Apply multisig plugin to the managementDAO
+  const applyTx = await pspContract.applyInstallation(
+    managementDAO.address,
+    {
+      helpersHash: hashHelpers(
+        installationPreparedEvent.preparedSetupData.helpers
+      ),
+      permissions: installationPreparedEvent.preparedSetupData.permissions,
+      plugin: installationPreparedEvent.plugin,
+      pluginSetupRef,
+    },
+    {
+      gasLimit: 30000000,
+    }
+  );
+  await applyTx.wait();
+
+  const multisigPluginPermission = {
+    operation: Operation.Grant,
+    where: {name: 'ManagementDAO', address: managementDAO.address},
+    who: {name: 'Multisig plugin', address: installationPreparedEvent.plugin},
+    permission: 'EXECUTE_PERMISSION',
+  };
+
+  const isPermissionCorrect = await isPermissionSetCorrectly(
+    managementDAO,
+    multisigPluginPermission
+  );
+
+  if (!isPermissionCorrect) {
+    const {who, where, operation} = multisigPluginPermission;
+    if (operation === Operation.Grant) {
+      throw new Error(
+        `(${who.name}: ${who.address}) doesn't have ${multisigPluginPermission.permission} on (${where.name}: ${where.address}) in ${managementDAO.address}`
+      );
+    }
+    throw new Error(
+      `(${who.name}: ${who.address}) has ${multisigPluginPermission.permission} on (${where.name}: ${where.address}) in ${managementDAO.address}`
+    );
+  }
```

Comment:
Very minor: I'd consider moving the if block into the function itself (`ensureGranted`, `ensureRevoked`), to make the main path more readable. A thrown error will interrupt execution anyway